src/main/java/ui/components/pickers/AssigneePickerDialog.java

```diff
@@ -0,0 +1,229 @@
+import java.util.stream.Collectors;
+
+public class AssigneePickerDialog extends Dialog<AssigneePickerDialogResponse> {
+    private static final String ID_TEXT_FIELD = "assigneePickerTextField";
+    private static final String TITLE_DIALOG = "Select Assignee";
+    private static final String TITLE_ASSIGNED_ASSIGNEE = "Assigned Assignee";
+    private static final String TITLE_ALL_ASSIGNEES = "All Assignees";
+    private static final String MESSAGE_NO_ASSIGNEE = "No assignee";
+    private static final String MESSAGE_NO_MATCHES = "No users matched your query.";
+
+    private final List<PickerAssignee> originalAssignees = new ArrayList<>();
+    private FlowPane assignedAssigneePane;
+    private VBox matchingAssigneesBox;
+    private TextField textField;
+    private AssigneePickerState state;
+
+    public AssigneePickerDialog(Stage stage, TurboIssue issue, List<TurboUser> assignees) {
+        initOwner(stage);
+        setTitle(TITLE_DIALOG);
+        setupButtons(getDialogPane());
+        setConfirmResultConverter();
+        originalAssignees.addAll(convertToPickerAssignees(issue, assignees));
+        state = new AssigneePickerState(originalAssignees);
+        initUI();
+        setupKeyEvents();
+        fillTextFieldWithExistingAssignee();
+    }
+
+    private void fillTextFieldWithExistingAssignee() {
+        PickerAssignee.getExistingAssignee(originalAssignees)
+                .map(PickerAssignee::getLoginName)
+                .ifPresent(this::fillTextFieldWithAssigneeLoginName);
+    }
+
+    private void fillTextFieldWithAssigneeLoginName(String assigneeLoginName) {
+        textField.setText(assigneeLoginName);
+    }
+
+    private void setupKeyEvents() {
+        textField.textProperty().addListener((observable, oldValue, newValue) -> {
+            handleUpdatedInput(newValue);
+        });
+    }
+
+    private void handleUpdatedInput(String userInput) {
+        state = new AssigneePickerState(originalAssignees, userInput);
+        refreshUI(state);
+    }
+
+    private void handleMouseClick(String assigneeName) {
+        textField.setText(assigneeName);
+        refreshUI(state);
+    }
+
+    private List<PickerAssignee> convertToPickerAssignees(TurboIssue issue, List<TurboUser> assignees) {
+        return assignees.stream()
+                .map(assignee -> {
+                    PickerAssignee convertedAssignee = new PickerAssignee(assignee);
+                    if (isExistingAssignee(issue, convertedAssignee)) {
+                        convertedAssignee.setExisting(true);
+                    }
+                    return convertedAssignee;
+                }).sorted().collect(Collectors.toList());
+    }
+
+    private boolean isExistingAssignee(TurboIssue issue, PickerAssignee assignee) {
+        if (!issue.getAssignee().isPresent()) return false;
+        return issue.getAssignee().get().equals(assignee.getLoginName());
+    }
+
+    private void setupButtons(DialogPane assigneePickerDialogPane) {
```

Comment:
To make this a little less verbose, we can maybe leave out the `assigneePicker` prefix, since it's clear from the context in which this method appears.